rackcity/views/rack_views.py

```diff
@@ -26,3 +30,66 @@ def rack_detail(request, pk):
     if request.method == 'GET':
         serializer = RackSerializer(rack)
         return JsonResponse(serializer.data)
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def rack_get(request):
+    """
+    List all racks within specified range.
+    """
+    range_serializer = RackRangeSerializer(data=request.data)
+    if range_serializer.is_valid():
+        try:
+            range_serializer.validate(request.data)
+        except Exception as error:
+            return JsonResponse(
+                {"failure_message": str(error)},
+                status=HTTPStatus.BAD_REQUEST
+            )
+    else:
+        return JsonResponse(
+            {"failure_message": "invalid keys or values"},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+
+    racks = Rack.objects.filter(
+        rack_num__range=range_serializer.get_number_range(request.data),
+        row_letter__range=range_serializer.get_row_range(request.data),
+    )
+    racks_with_instances = []
+    for rack in racks:
+        rack_serializer = RackSerializer(rack)
+        instances = ITInstance.objects.filter(rack=rack.id)
+        instances_serializer = ITInstanceSerializer(
+            instances,
+            many=True
+        )
+        rack_detail = {
+            "rack": rack_serializer.data,
+            "instances": instances_serializer.data,
+        }
+        racks_with_instances.append(rack_detail)
+
+    return JsonResponse(
+        {"racks": racks_with_instances},
+        status=HTTPStatus.OK
+    )
+
+
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def rack_create(request):
```

Comment:
Sorry to be picky but can we make these return `JsonResponse({}, status=5xx)` until they're implemented in case Brendan or Claire aren't tracking